src/WasmExecutor.cpp

```diff
@@ -0,0 +1,1583 @@
+        val.Set(*(const T *)slot);
+    }
+};
+
+template<>
+inline void LoadAndReturnScalar<void*>::operator()(const Local<Context> &context, const void *slot, ReturnValue<Value> val) {
+    internal_error << "TODO: 64-bit slots aren't yet supported";
+}
+
+template<>
+inline void LoadAndReturnScalar<uint64_t>::operator()(const Local<Context> &context, const void *slot, ReturnValue<Value> val) {
+    internal_error << "TODO: 64-bit slots aren't yet supported";
+}
+
+template<>
+inline void LoadAndReturnScalar<int64_t>::operator()(const Local<Context> &context, const void *slot, ReturnValue<Value> val) {
+    internal_error << "TODO: 64-bit slots aren't yet supported";
+}
+
+// ------------------------------
+
+template<typename T>
+struct WrapScalar {
+    Local<Value> operator()(const Local<Context> &context, const void *val_ptr) {
+        double val = *(const T *)(val_ptr);
+        Isolate *isolate = context->GetIsolate();
+        return Number::New(isolate, val);
+    }
+};
+
+template<>
+inline Local<Value> WrapScalar<void*>::operator()(const Local<Context> &context, const void *val_ptr) {
+    internal_error << "TODO: 64-bit slots aren't yet supported";
+    return Local<Value>();
+}
+
+template<>
+inline Local<Value> WrapScalar<uint64_t>::operator()(const Local<Context> &context, const void *val_ptr) {
+    internal_error << "TODO: 64-bit slots aren't yet supported";
+    return Local<Value>();
+}
+
+template<>
+inline Local<Value> WrapScalar<int64_t>::operator()(const Local<Context> &context, const void *val_ptr) {
+    internal_error << "TODO: 64-bit slots aren't yet supported";
+    return Local<Value>();
+}
+
+// ------------------------------
+
+Local<Value> wrap_scalar(const Local<Context> &context, const Type &t, const void *val_ptr) {
+    return dynamic_type_dispatch<WrapScalar>(t, context, val_ptr);
+}
+
+template<typename T>
+Local<Value> wrap_scalar(const Local<Context> &context, const T &val) {
+    return WrapScalar<T>()(context, &val);
+}
+
+// ---------------------------------
+
+enum EmbedderDataSlots {
+    // don't use slot 0
+    kWasmMemoryObject = 1,
+    kBDMallocPtr,
+    kHeapBase,
+    kJitUserContext,
+    kString_buffer,
+    kString_grow,
+};
+
+wasm32_ptr_t v8_WasmMemoryObject_malloc(const Local<Context> &context, size_t size) {
+    Isolate *isolate = context->GetIsolate();
+
+    BDMalloc *bdmalloc = (BDMalloc *) context->GetAlignedPointerFromEmbedderData(kBDMallocPtr);
+    if (!bdmalloc->inited()) {
+        int32_t heap_base = context->GetEmbedderData(kHeapBase)->Int32Value(context).ToChecked();
+
+        Local<Object> memory_value = context->GetEmbedderData(kWasmMemoryObject).As<Object>();  // really a WasmMemoryObject
+        Local<Object> buffer_string = context->GetEmbedderData(kString_buffer).As<Object>();
+        Local<ArrayBuffer> wasm_memory = Local<ArrayBuffer>::Cast(memory_value->Get(buffer_string));
+
+        wdebug(0)<<"heap_base is: "<<heap_base<<"\n";
```

Comment:
Please add spaces around << in this file

Reply:
A lot of the debugging in here started as quick-hack-for-debugging and then migrated to leave-in-long-term; I should have done more cleaning. Done.